### PR TITLE
Update marking criteria, close #121

### LINKED ISSUE #121: Changes for 2024-25
- [ ] Assessment criteria: currently there's not much on code quality: conciseness, consistency of code? https://github.com/ITSLeeds/TDS/blob/master/coursework-template.md#marks
- [ ] 2 pager: 2 pages is too short for some people, bump up to 3?
- [ ] Any other suggestions @mem48 and @yyuanxuan?

### PR DESCRIPTION
**Pull request by Robinlovelace**

### CODE REVIEW

File: coursework-template.Rmd [hunk 1]
```diff
@@ -179,37 +178,72 @@ You could pick one of these topics:
...
+1. Adding value to datasets with joins (key-based or spatial), creation of new variables (also known as feature engineering) and reshaping data (e.g. from wide to long format)
 
-    **Choice of topic and impact**
+**Distinction (70%+):** The report makes use of a complex (with many columns and rows) and/or multiple input datasets, efficiently importing them and adding value by creating new variables, recategorising, changing data formats/types, and/or reshaping the data. Selected datasets are appropriate for the research questions, clearly described, with links to the source and understanding of how the datasets were generated.
 
-1. Topic selection, including originality, availability of datasets related to the topic and relevance to solving transport planning problems
-1. Discuss alternative topics considered and justify the topic
-1. Make specific recommendations for further research or policies that are supported by the data analysis undertaken
-1. Explain the potential impacts of the work
+**Merit (60-69%):** Between pass and distinction.
```
Comment (mem48) [inline coursework-template.Rmd:196]: This is not very specific. Doesn't help students or markers to know where the threshold is.

File: coursework-template.Rmd [hunk 2]
```diff
@@ -179,37 +178,72 @@ You could pick one of these topics:
...
+
+**Pass (50-59%):** The selection, description of, cleaning and value-added to the input datasets are weak overall. The report does not make use of complex or multiple input datasets, or the datasets are not appropriate for the research questions, or the datasets are not clearly described, or there are no links to the source or understanding of how the datasets were generated.
+
+**Fail (0-49%):** The report does not make use of appropriate input datasets and contains very little or now evidence of data cleaning, adding value to the datasets or reshaping the data.
```
Comment (mem48) [inline coursework-template.Rmd:200]: This is negative description. I.e. just says don't meet pass criteria. Need some positive description of what the student can do and still fail.

File: coursework-template.Rmd [hunk 3]
```diff
@@ -179,37 +178,72 @@ You could pick one of these topics:
...
 
-1. Topic selection, including originality, availability of datasets related to the topic and relevance to solving transport planning problems
-1. Discuss alternative topics considered and justify the topic
-1. Make specific recommendations for further research or policies that are supported by the data analysis undertaken
-1. Explain the potential impacts of the work
+**Merit (60-69%):** Between pass and distinction.
+
+**Pass (50-59%):** The selection, description of, cleaning and value-added to the input datasets are weak overall. The report does not make use of complex or multiple input datasets, or the datasets are not appropriate for the research questions, or the datasets are not clearly described, or there are no links to the source or understanding of how the datasets were generated.
```
Comment (mem48) [inline coursework-template.Rmd:198]: This sounds like a fail description but marked as a pass. The things you describe sound to me like the basics. So why would a student get a passing grade? Doesn't seem to be anywhere in the markscheme for mistakes. All do/don't but no attempted, or limited effort.

File: coursework-template.Rmd [hunk 4]
```diff
@@ -179,37 +178,72 @@ You could pick one of these topics:
...
-1. High quality, attractive and innovative visualisations
-1. Description of visualisation techniques, e.g. with reference to particular packages, commands, or the literature on how to produce effective data visualisations
+1. High quality, attractive or advanced techniques (e.g. multi-layered maps or graphs, facets or other advanced techniques)
 1. Using visualisation techniques appropriate to the topic and data and interpreting the results correctly (e.g. mentioning potential confounding factors that could account for observed patterns)
+1. The report is well-formatted, accessible (e.g. with legible text size and does not contain excessive code in the submitted report) and clearly communicates the data and analysis visually, with appropriate figure captions, cross-references and a consistent style
+
+**Distinction (70%+):** The report contains high quality, attractive, advanced and meaningful visualisations that are well-described and interpreted correctly. The report is well-formatted, accessible and clearly communicates the data and analysis visually.
+
+**Merit (60-69%):** Between pass and distinction.
```
Comment (mem48) [inline coursework-template.Rmd:211]: Again too vauge

File: coursework-template.Rmd [hunk 5]
```diff
@@ -179,37 +178,72 @@ You could pick one of these topics:
...
+1. The report is fully reproducible, including generation of figures. There are links to online resources for others wanting to reproduce the analysis for another area, and links to the input data
+
+**Distinction (70%+):** The source code underlying the report contains high quality, efficient and reproducible code that is well-written, using consistent syntax and good style, well-commented and uses appropriate packages. The report is fully reproducible, with links to online resources for others wanting to reproduce the analysis for another area, and links to the input data.
+
+**Merit (60-69%):** Between pass and distinction.
+
+**Pass (50-59%):** The source code underlying the report is available and describes the outputs in the report but is not well-commented, not efficient or has very limited levels of reproduicibility, with few links to online resources for others wanting to reproduce the analysis for another area, and few links to the input data.
+
+**Fail (0-49%):** The report has little to no reproducible, readable or efficient code.
```
Comment (mem48) [inline coursework-template.Rmd:229]: So a student that did a first class piece of work but forgot to submit the code would scrape a pass. 0.8 × 70% = 56%? In clear on how to award grades between 0 and 49%.

File: coursework-template.Rmd [hunk 6]
```diff
@@ -179,37 +178,72 @@ You could pick one of these topics:
...
+1. Use of appropriate data science methods and techniques
+1. Discussion of the strengths and weaknesses of the analysis and input datasets and/or how limitations could be addressed
+1. Discuss further research and/or explain the potential impacts of the work
+1. The conclusions are supported by the analysis and results
+1. The contents of the report fit together logically and support the aims and/or research questions of the report
+
+**Distinction (70%+):** The report contains a clear research question, appropriate reference to the academic, policy and/or technical literature, use of appropriate data science methods and techniques, discussion of the strengths and weaknesses of the analysis and input datasets and/or how limitations could be addressed, discussion of further research and/or explanation of the potential impacts of the work, conclusions supported by the analysis and results, and the contents of the report fit together logically and support the aims and/or research questions of the report.
 
-    **Understand data science processes, concepts, methods and approaches**
+**Merit (60-69%):** Between pass and distinction.
```
Comment (mem48) [inline coursework-template.Rmd:244]: Again too vauge

File: coursework-template.Rmd [hunk 7]
```diff
@@ -179,37 +178,72 @@ You could pick one of these topics:
...
-1. Enable reproducibility, by documenting key parts the code have written as part of the analysis, e.g. with reference to a key code chunk shown in the report and with reference to particular lines of code in the RMarkdown file underpinning the analysis, links to online resources for others wanting to reproduce the analysis for another area, and links to the input data, the report is fully reproducible, including generation of figures where possible and guidance on how to reproduce the results (e.g. provided in a footnote in the report or with a link to code stored in a platform such as GitHub/GitLab)
-1. Discussion of the limitations of the analysis and implications for the findings and how they could be addressed in future research and what type of data collection activities could address those limitations
+**Pass (50-59%):** The report contains a research question but only limited references to appropriate literature, the analysis is weak, with limitted understandings of the limitations of the results, conclusions are not supported by the analysis and results or the report does not fit together well in a cohesive whole.
```
Comment (mem48) [inline coursework-template.Rmd:246]: No fail description?

Comment (mem48) [inline coursework-template.Rmd:246]: Again this sounds like a fail (have a research question but otherwise rubbish)

**Comment by Robinlovelace** [comment]
Can you have a bash at these @mem48 ?

File: coursework-template.Rmd (hunk 1 above)
Comment (Robinlovelace) [reply coursework-template.Rmd:196]: Added: 

> **Merit (60-69%):** The report makes some use of complex or multiple input datasets. The selection, description of, cleaning or value-added to the input datasets show skill and care applied to the data processing stage but with some weaknesses. Selected datasets are appropriate for the research questions, with some description or links to the data source.

File: coursework-template.Rmd (hunk 3 above)
Comment (Robinlovelace) [reply coursework-template.Rmd:198]: **Pass (50-59%):** There is some evidence of care and attention put into the selection, description of or cleaning of the input datasets but little value has been added. The report makes little use of complex or multiple input datasets. The datasets are not appropriate for the research questions, the datasets are not clearly described, or there are no links to the source or understanding of how the datasets were generated, but the data processing aspect of the work acceptable.

File: coursework-template.Rmd (hunk 2 above)
Comment (Robinlovelace) [reply coursework-template.Rmd:200]: > While there may be some evidence of data processing, it is of poor quality and/or not appropriate for the research questions.

Added.

**Comment by Robinlovelace** [comment]
Updates added @mem48 do take a look, any further comments/suggestions welcome.

**Pull request merged**